src/v/cloud_topics/batcher/tests/batcher_test.cc

```diff
@@ -79,26 +85,165 @@ get_random_reader(int num_batches, int num_records) { // NOLINT
     };
 }
 
+namespace experimental::cloud_topics {
+struct batcher_accessor {
+    ss::future<result<bool>> run_once() noexcept { return batcher->run_once(); }
+
+    // Returns true if the write request is in the `_pending` collection
+    bool write_requests_pending(size_t n) {
+        return batcher->_pending.size() == n;
+    }
+
+    cloud_topics::batcher<ss::manual_clock>* batcher;
+};
+} // namespace experimental::cloud_topics
+
+ss::future<> sleep(std::chrono::milliseconds delta, int retry_limit = 100) {
+    ss::manual_clock::advance(delta);
+    for (int i = 0; i < retry_limit; i++) {
+        co_await ss::yield();
+    }
+}
+
+// Simulate sleep of certain duration and wait until the condition is met
+template<class Fn>
+ss::future<>
+sleep_until(std::chrono::milliseconds delta, Fn&& fn, int retry_limit = 100) {
+    ss::manual_clock::advance(delta);
+    for (int i = 0; i < retry_limit; i++) {
+        co_await ss::yield();
+        if (fn()) {
+            co_return;
+        }
+    }
+    GTEST_MESSAGE_("Test stalled", ::testing::TestPartResult::kFatalFailure);
+}
+
+inline cloud_topics::dl_placeholder
```

Comment:
inline is unnecessary

Reply:
merging as is and will change that in a followup